### ISSUE TITLE
Disney+ and new ISA v21.4.6 - KODI hangs or crashes

### ISSUE THREAD

**Issue by zuzia-dev** [matthuisman/slyguy.addons#731]
System: Debian GNU/Linux 12 (bookworm) x86_64
InputStream Adaptive: v21.4.6

The problem is with both codecs HEVC and AVC.
```
2024-05-02 13:59:46.392 T:728910    info <general>: script.module.slyguy - Python Version: 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]
2024-05-02 13:59:46.399 T:728910    info <general>: script.module.slyguy - Proxy Started: http://127.0.0.1:37321/
2024-05-02 13:59:46.402 T:728910    info <general>: script.module.slyguy - Widevine Level (fallback): 3
2024-05-02 13:59:46.403 T:728910    info <general>: script.module.slyguy - HDCP Level (fallback): 0.0
2024-05-02 14:01:55.836 T:728969    info <general>: slyguy.disney.plus - Widevine - System: Linux | Arch: x86_64
2024-05-02 14:01:56.114 T:728969    info <general>: slyguy.disney.plus - Widevine Current MD5: cba9c2f7f354d70c2120f864c930db5d
2024-05-02 14:01:57.473 T:728865    info <general>: VideoPlayer::OpenFile: plugin://slyguy.disney.plus/?_=play&_play=1&content_id=9d638ea3-49bd-442a-b4a3-03b7debea6db&profile_id=c436b494-3758-4ea3-90d6-ea535bbcbc12
2024-05-02 14:01:57.479 T:729101    info <general>: Creating InputStream
2024-05-02 14:01:58.220 T:729101 warning <general>: AddOnLog: inputstream.adaptive: Warning "inputstream.adaptive.manifest_type" property is deprecated and will be removed next Kodi version, the manifest type is now automatically detected.
                                                   If you are using a proxy remember to add the appropriate "content-type" header to the HTTP manifest response
                                                   See Wiki page "How to provide custom manifest/license" to learn more about it.
2024-05-02 14:01:58.233 T:729101    info <general>: AddOnLog: inputstream.adaptive: [Repr. chooser] Resolution set: 1920x1080, max allowed: 1920x1080, Adjust refresh rate: 0
2024-05-02 14:01:58.630 T:728865    info <general>: Loading skin file: DialogSelect.xml, load type: KEEP_IN_MEMORY
2024-05-02 14:02:01.429 T:729101 warning <general>: AddOnLog: inputstream.adaptive: ParseMultivariantPlaylist: Unknown encryption type
2024-05-02 14:02:01.452 T:729101    info <general>: Skipped 1 duplicate messages..
2024-05-02 14:02:01.452 T:729101   error <general>: AddOnLog: inputstream.adaptive: ParseMultivariantPlaylist: Cannot find variant for AUDIO GROUP-ID: aac-64k
2024-05-02 14:02:01.490 T:729101    info <general>: Skipped 11 duplicate messages..
2024-05-02 14:02:01.490 T:729101    info <general>: AddOnLog: inputstream.adaptive: Manifest successfully parsed (Periods: 1, Streams in first period: 69, Type: live)
2024-05-02 14:02:01.495 T:729101    info <general>: Creating Demuxer
2024-05-02 14:02:01.607 T:729101    info <general>: Opening stream: 1001 source: 256
```

**Comment by matthuisman** [comment]
that appears to be an issue in inputstream.adaptive.
I would recommend opening a bug report there

**Comment by zuzia-dev** [comment]
@matthuisman Thanks for your suggestion. Problem solved: https://github.com/xbmc/inputstream.adaptive/pull/1547

**Comment by matthuisman** [comment]
as above, the latest Inputstream Adaptive version broke Disney+.
There is a PR up to fix it but that could take some time (1 week-ish?) to get merged and released
https://github.com/xbmc/inputstream.adaptive/pull/1547

The other option is to revert to an older Inputstream Adaptive version (v21.4.5)

**Comment by zuzia-dev** [comment]
A version with the patch applied is available: https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-1547/2/artifacts
For Linux systems (zip file): https://github.com/zuzia-dev/Kodi-Omega-addons-for-Linux/commit/18915849a2a14f79e420332f054a9f77b9a62e78

**Comment by zuzia-dev** [comment]
The InputStream Adaptive v21.4.7 version is now available. The problem with Disney+ has been resolved.

**Comment by matthuisman** [comment]
Yah. Closing this ticket as fixed in 21.4.7
Builds here: https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/21.4.7-Omega/1/artifacts
